micro_g/scripts/pose_servoing_controller.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/python3
+# Copyright 2023, Micro-G Dev Team
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import argparse
+import sys
+import time
+
+import numpy as np
+import rclpy
+import tf2_ros
+import transforms3d
+from geometry_msgs.msg import PoseStamped
+from interbotix_common_modules.angle_manipulation import angle_manipulation as ang
+from interbotix_xs_modules.xs_robot.arm import InterbotixManipulatorXS
+from rclpy.utilities import remove_ros_args
+from tf2_geometry_msgs import do_transform_pose
+
+
+class XSArmPoseServoingController(InterbotixManipulatorXS):
+    """
+    Tracks a published end effector pose.
+    """
+
+    def __init__(
+        self,
+        robot_model: str,
+        robot_name: str,
+        control_update_rate: float = 10.0,
+        moving_time: float = 0.5,
+        xs_args=None,
+    ):
+        """
+        Create the controller
+
+        Args:
+            robot_model: the robot model
+            robot_name: the robot name
+            control_update_rate: the rate at which the controller is updated (Hz).
+            moving_time: the duration in which all movements should take place (s).
+            xs_args: Extra arguments to be passed to InterbotixManipulatorXS. Defaults
+                to None.
+        """
+        # Initialize the interface to the arm
+        InterbotixManipulatorXS.__init__(
```

Comment:
What's the motivation for using this instead of `super`?

Reply:
The motivation is that I didn't understand `super` :sweat:, but I've read up on it and changed this to use `super`.